raiden/utils/claim.py

```diff
@@ -137,32 +147,47 @@ def get_state_changes_for_claims(
                 settle_transaction=None,
             )
 
-            state_changes.append(
-                ContractReceiveChannelNew(
-                    channel_state=channel_state,
-                    transaction_hash=TransactionHash(b""),
-                    block_number=BlockNumber(0),
-                    block_hash=BlockHash(b""),
-                )
-            )
-            state_changes.append(
-                ContractReceiveChannelDeposit(
-                    canonical_identifier=CanonicalIdentifier(
-                        chain_identifier=claim.chain_id,
-                        token_network_address=claim.token_network_address,
-                        channel_identifier=claim.channel_id,
+            state_changes.extend(
+                [
+                    ContractReceiveChannelNew(
+                        channel_state=channel_state,
+                        transaction_hash=TransactionHash(b""),
+                        block_number=BlockNumber(0),
+                        block_hash=BlockHash(b""),
                     ),
-                    deposit_transaction=TransactionChannelDeposit(
-                        participant_address=claim.owner,
-                        contract_balance=claim.total_amount,
-                        deposit_block_number=BlockNumber(1),
-                        claim=claim,
+                    # Fake the deposit with the claims value
+                    ContractReceiveChannelDeposit(
+                        canonical_identifier=CanonicalIdentifier(
+                            chain_identifier=claim.chain_id,
+                            token_network_address=claim.token_network_address,
+                            channel_identifier=claim.channel_id,
+                        ),
+                        deposit_transaction=TransactionChannelDeposit(
+                            participant_address=claim.owner,
+                            contract_balance=claim.total_amount,
+                            deposit_block_number=BlockNumber(1),
+                            claim=claim,
+                        ),
+                        transaction_hash=TransactionHash(b""),
+                        block_number=BlockNumber(1),
+                        block_hash=BlockHash(b""),
+                        fee_config=MediationFeeConfig(),
                     ),
-                    transaction_hash=TransactionHash(b""),
-                    block_number=BlockNumber(1),
-                    block_hash=BlockHash(b""),
-                    fee_config=MediationFeeConfig(),
-                )
+                    # Fake the already withdrawn amount
+                    ContractReceiveChannelWithdraw(
```

Comment:
I'm a bit lost: Do we also need to consider burned amounts here somehow?

Reply:
I think we do not need to consider burnt amounts here as we store it in the state for each participant. Every burnt amount on chain must have a burn proof signed by the client. So the client should be aware of it anyway.

Reply:
If we receive a deposit event in vanilla raiden, do we need to take withdrawn amount with that state change into account?

Reply:
> I think we do not need to consider burnt amounts here as we store it in the state for each participant.

What would happen if the state would get lost?